mysql/mysql.go

```diff
@@ -66,8 +69,13 @@ func ConfigFromURL(u *url.URL) (cfg *mysql.Config, err error) {
 	} else {
 		cfg = mysql.NewConfig()
 	}
-	cfg.Net = "tcp"
-	cfg.Addr = u.Host
+	if matches := netAddrRE.FindStringSubmatch(u.Host); len(matches) == 3 {
```

Comment:
Can we include this stuff from u.Host in the string we pass to muysql.ParseDSN above instead of doing our own parsing?

Reply:
`mysql.ParseDSN` expects the host to be in the form of `protocol(address)` so simply passing `mysql://localhost/db` won't work so we'll need to wrap the host with `'('  + u.Host  +')'`. In any case we'll need to add our own logic.